### DISCUSSION TITLE
Error while running a validation application on React.js

### DISCUSSION THREAD

**Discussion by hussainshahzad5471** [iTwin/community#43] [General]
When running the validation application on React.js which i got from https://github.com/iTwin/Course-Property-Validation-API.

the URL : https://api.bentley.com/imodel/rpc//mode/1/context/b997c9fe-eb6d-4fdb-907b-3313e7591dde/imodel/5e8e0bca-ac64-47bb-b93a-3933e30339b6/changeset/0/IModelReadRpcInterface-3.1.0-getConnectionProps

response: {"error":"client spa-aV2KgrBuko9PIBZWS1u4m5frm is not allowed to use visualization version v3"}

**Comment by aruniverse** [comment]
This course and the guide (the repo) is using iTwin.js 3.x which has just recently reached end of life. The code needs to be updated to use the latest 4.x version, which shouldn't be a huge effort. You can see the migration guide [here](https://github.com/iTwin/itwinjs-core/discussions/5606) if you want to try it locally. We will look to getting the repo updated in the meanwhile
